### PR TITLE
[jit] Move Python 3 tests to their own file

### PR DESCRIPTION
**Pull request by driazati**
Keeping them in their own file lets us use Python3 syntax directly instead of going through the string frontend

This also deletes an assert added in #13250, cc @ezyang do you know why this check was added in the first place?

The test cases defined outside `test_jit.py` work fine with Python's unittest runner. Pytest doesn't find them though, so to use Pytest for those files you have to run the file directly instead of doing `pytest test_jit.py`

Differential Revision: [D15704079](https://our.internmc.facebook.com/intern/diff/15704079/)

### CODE REVIEW

File: test/test_jit.py [hunk 1]
```diff
@@ -5154,26 +5098,9 @@ def fn():
...
-            raise Exception("hello")
-        ast = torch.jit.frontend.get_jit_def(fn)
-        self.assertExpected(str(ast))
-
     def _make_scalar_vars(self, arr, dtype):
         return [torch.tensor(val, dtype=dtype) for val in arr]
 
-
-    @unittest.skipIf(PY2, "tuple printing in py2 is different than torchscript")
-    def test_string_print(self):
```
Comment (eellison) [inline test/test_jit.py]: This was deleted

File: test/test_jit.py [hunk 2]
```diff
@@ -5154,26 +5098,9 @@ def fn():
         ast = torch.jit.frontend.get_jit_def(fn)
         self.assertExpected(str(ast))
 
-    @unittest.skipIf(PY2, "Requires python 3")
-    def test_python_frontend_py3(self):
```
Comment (eellison) [inline test/test_jit.py]: Also deleted

File: test/test_jit.py [hunk 3]
```diff
@@ -3518,23 +3479,6 @@ def func(a, b):
         b = torch.rand(1, requires_grad=True)
         self.checkScript(func, (a, b), optimize=True)
 
-    @unittest.skipIf(not PY35, "Python 3.5 needed")
-    def test_matmul_py3(self):
-        code = dedent("""
-        def fn(a, b):
```
Comment (eellison) [inline test/test_jit.py]: Deleted

File: test/test_jit.py (hunk 1 above)
Comment (driazati) [reply test/test_jit.py]: It's here https://github.com/pytorch/pytorch/pull/21489/files#diff-ff4f8670281cd1eb4e09329cc1dcb43bR50

File: test/test_jit.py (hunk 2 above)
Comment (driazati) [reply test/test_jit.py]: here https://github.com/pytorch/pytorch/pull/21489/files#diff-ff4f8670281cd1eb4e09329cc1dcb43bR41

File: test/test_jit.py (hunk 3 above)
Comment (driazati) [reply test/test_jit.py]: it's moved, the `_py3` was removed

File: test/common_utils.py [hunk 4]
```diff
@@ -977,16 +977,6 @@ def set_running_script_path():
         pass
 
 
-def check_test_defined_in_running_script(test_case):
```
Comment (eellison) [inline test/common_utils.py]: Why was this removed ?

**Review by eellison — APPROVED** [review]
Looks good but i think you need FBCode changes so the test runs internally

**Review by suo — APPROVED** [review]

**Comment by ezyang** [comment]
The assert exists because people sometimes imported `test_foo` from `test_bar`, and the net effect of this is that the foo tests run twice: once when we run `test_foo.py`, and then another time in `test_bar.py`. I'd prefer if you did not remove this assert, thanks!

**Review by ezyang — CHANGES_REQUESTED** [review]
Please figure out why the assert was twinging.
